doc/Programming-guide/Your-first-Kaa-application/index.md

```diff
@@ -2,7 +2,615 @@
+membership. Notifications can be sent either to every endpoint subscribed to a topic
+or to an individual endpoint.
+
+Notifications can be assigned expiration timestamps to prevent their delivery after a certain period of time.
+
+For programming practice, [see using notifications]({{root_url}}/Programming-guide/Key-platform-features/Notifications/).
+
+
+## Events
+
+Kaa allows for delivery of _events_, which are structured messages, across endpoints.
+When endpoints register with the Kaa server, they communicate which event types
+they are able to generate and receive. Kaa allows endpoints to send events either
+to virtual “chat rooms” or to individual endpoints.
+
+Events can even be delivered across applications registered with Kaa – making it possible to quickly integrate and enable interoperability between endpoints running different applications
+
+Some examples are: a mobile application that controls house lighting,
+a car’s GPS that communicates with the home security system, a set of integrated
+audio systems from different vendors that deliver a smooth playback experience as you walk
+from one room to another.
+
+Kaa events are implemented in a generic, abstract way, using non-proprietary schema
+definitions that ensure identical message structures. The schema provides independence
+from any specific functionality implementation details.
+
+For programming practice, see [messaging across endpoints]({{root_url}}/Programming-guide/Key-platform-features/Events/).
+
+## Distributing operational data
+
+Kaa allows you to perform operational data updates, such as configuration data updates,
+from the Kaa server to endpoints.
+This feature can be used for centralized configuration management, content distribution, etc.
+Since Kaa works with structured data and constraint types, it guarantees data integrity.
+
+The Kaa server monitors the database for changes and distributes updates to endpoints
+in the incremental form, thus ensuring efficient bandwidth use.
+The endpoint SDK performs data merging and persistence, as well as notifies the client code
+about the specific changes made to the data.
+As a result, the client application knows exactly where in the data structure the changes
+occurred and can be programmed to react accordingly.
+
+Based on the endpoint’s group membership, it is possible to control what data
+is available to the endpoint. This is achieved by applying group-specific data overrides,
+which make it possible to adjust the behavior of the client application based on operational
+conditions or usage patterns, fine-tune the algorithms according to feedback,
+implement gradual feature roll-out, A/B testing, etc.
+
+For programming practice, see [LINK DOESN'T EXIST YET : distributing data to endpoints](http://docs.kaaproject.org/display/KAA/Distributing+data+to+endpoints)
+
+# Troubleshooting
+
+To Be Defined.
```

Comment:
Is it intended for this section to be empty for now?
If yes, I'd better drop the section.